File: backend/app/Enums/MessageType.php
<?php

namespace App\Enums;

/**
 * messages.type. `text` (Phase 1 item 8), `voice_note` (Phase 3 item 1),
 * `gif` (Phase 3 item 2) and `photo` (Phase 3 item 3) are all written now —
 * every case in this enum is in active use. Declared in full from the start
 * anyway so no migration was ever needed as each of these got confirmed in
 * scope.
 */
enum MessageType: string
{
    case Text = 'text';
    case VoiceNote = 'voice_note';
    case Gif = 'gif';
    case Photo = 'photo';
}

File: mobile/lib/chat/domain/message_type.dart
/// Mirrors `App\Enums\MessageType` on the backend. [text] and, since Phase 3
/// item 1, [voiceNote] are sent by this feature — gif/photo are open
/// decisions #17-18, still unconfirmed [TBD-17/18] — but the full enum is
/// declared now so an unrecognised value from a future server (or another
/// client) never crashes [MessageType.fromApiValue], and no migration is
/// needed if/when the rest are confirmed in scope.
enum MessageType {
  text('text'),
  voiceNote('voice_note'),
  gif('gif'),
  photo('photo');

  const MessageType(this.apiValue);

  final String apiValue;

  factory MessageType.fromApiValue(String value) => MessageType.values
      .firstWhere((t) => t.apiValue == value, orElse: () => MessageType.text);
}
